testing/tests/DevExpress.ui.widgets/treeViewParts/focusing.js

```diff
@@ -238,6 +241,86 @@ QUnit.test('First node should not has been focused when focusing on SelectAll it
     }
 });
 
+QUnit.test('SelectAll item should be focused when focusing a treeview if showCheckBoxesMode:selectAll', function(assert) {
+    if(!isDesktopDevice()) {
+        assert.ok(true, 'only on desktops');
+        return;
+    }
+
+    const $treeView = initTree({
+        items: [ { id: 1 } ],
+        showCheckBoxesMode: 'selectAll'
+    });
+
+    $treeView.trigger('focusin');
+
+    assert.ok($(`.${SELECT_ALL_ITEM_CLASS}`).hasClass(FOCUSED_STATE_CLASS));
+});
+
+QUnit.test('SelectAll item should be focused when focusing a treeview second time if showCheckBoxesMode:selectAll', function(assert) {
+    if(!isDesktopDevice()) {
+        assert.ok(true, 'only on desktops');
+        return;
+    }
+
+    const $treeView = initTree({
+        items: [ { id: 1 } ],
+        showCheckBoxesMode: 'selectAll',
+    });
+
+    const $selectAllItem = $(`.${SELECT_ALL_ITEM_CLASS}`);
+
+    const clock = sinon.useFakeTimers();
+
+    try {
+        $treeView.trigger('focusin');
+        $treeView.trigger('focusin');
+        clock.tick(10);
+
+        assert.ok($selectAllItem.hasClass(FOCUSED_STATE_CLASS));
+    } finally {
+        clock.restore();
+    }
+});
+
+QUnit.test('First node should be focused when focusing a treeview if showCheckBoxesMode:normal', function(assert) {
+    if(!isDesktopDevice()) {
+        assert.ok(true, 'only on desktops');
+        return;
+    }
+
+    const $treeView = initTree({
+        items: [ { id: 1 }, { id: 2 } ],
+        showCheckBoxesMode: 'normal',
+    });
+
+    const $firstItem = $(`.${NODE_CLASS}`).first();
+
+    $treeView.trigger('focusin');
+
+    assert.ok($firstItem.hasClass(FOCUSED_STATE_CLASS));
+});
+
+QUnit.test('SelectAll checkbox should be checked with space key', function(assert) {
+    if(!isDesktopDevice()) {
+        assert.ok(true, 'only on desktops');
+        return;
+    }
+
+    const $treeView = initTree({
+        items: [ { id: 1 }],
+        showCheckBoxesMode: 'selectAll',
+    });
+
+    const $selectAllItem = $(`.${SELECT_ALL_ITEM_CLASS}`);
+
+    $selectAllItem.trigger('focusin');
+
+    keyboardMock($treeView).keyDown('space');
+
+    assert.ok($selectAllItem.hasClass(CHECKBOX_CHECKED_STATE_CLASS));
+});
+
```

Comment:
**Scenario**: focus treeView with `showCheckBoxesMode: "selectAll"`. Press `tab`. 
**Actual behavior**: "selectAll" checkboxes is focused again (but now you can navigate down)
**Expected behavior**: focus is moved away from treeView